### PR TITLE
OCPBUGS-11930: Clean up existing VPC endpoint connections

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
When authentication with a customer account fails due to an OIDC provider no longer being available, the HyperShift operator should unilaterally cleanup any VPC endpoint services that exist on the management side. In some cases, this cannot happen until any existing private link connections have been terminated. This commit adds code to find any existing connections and reject them so that the endpoint service can be removed.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #OCPBUGS-11930

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Review by bryan-cox — APPROVED** [review]
/lgtm

File: hypershift-operator/controllers/platform/aws/controller.go [hunk 1]
```diff
@@ -521,10 +521,36 @@ func (r *AWSEndpointServiceReconciler) delete(ctx context.Context, awsEndpointSe
 		ServiceIds: []*string{aws.String(serviceID)},
 	})
 	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
-			return false, errors.New(awsErr.Code())
+		log.Error(err, "Failed to delete vpc endpoint service configuration", "serviceID", serviceID)
```
Comment (enxebre) [inline hypershift-operator/controllers/platform/aws/controller.go:524]: to reduce the number of code paths and complexity, we could also do this unconditionally not only if error?

**Comment by enxebre** [comment]
/lgtm
/hold
https://github.com/openshift/hypershift/pull/2438#discussion_r1171505643
feel free to cancel

File: hypershift-operator/controllers/platform/aws/controller.go (hunk 1 above)
Comment (csrwng) [reply hypershift-operator/controllers/platform/aws/controller.go:524]: In the "happy" path (and even not so happy path), we normally don't need to do this. I didn't want to add the additional calls unless strictly necessary.

**Pull request merged**